partner_communication/models/communication_job.py

```diff
@@ -345,6 +345,12 @@ def send(self):
         to_print = todo.filtered(lambda j: j.send_mode == 'physical')
         for job in todo.filtered(lambda j: j.send_mode in ('both',
                                                            'digital')):
+            origin = self.env.context.get('origin')
+            # if we print first in a communication with send_mode == both
+            if origin == "both_print" and job.send_mode == 'both':
+                job.send_mode = 'digital'
+                return job._print_report()
```

Comment:
This will update the send_mode to `done` so you might want to call _print_report before setting the send_mode to digital.

Reply:
I modified a bit the print_report function to especially correct that (with the origin parameter in context)